src/gleam/string.gleam

```diff
@@ -739,3 +739,70 @@ pub fn to_option(s: String) -> Option(String) {
     _ -> Some(s)
   }
 }
+
+/// Returns the first element in a grapheme and wraps it in an `Option(String)`.
+/// If the `String` is empty, it returns `None`. Otherwise, it returns
+/// `Some(String)`.
+///
+/// ## Examples
+///
+/// ```gleam
+/// > first("")
+/// None
+/// ```
+///
+/// ```gleam
+/// > first("icecream")
+/// Some("i")
+/// ```
+///
+pub fn first(s: String) -> Option(String) {
+  case length(s) {
```

Comment:
Length is quite an expensive operation in this case because it has to traverse the entire string to count the number of graphemes. We will want to only check the start of the string to see if it is long enough.